### PR TITLE
CDAP-19638: add fqn support for batch plugins

### PR DESCRIPTION
**Pull request by mrahanjam**
This PR adds FQN to the Salesforce batch plugin code.

Jira: [CDAP-19638](https://cdap.atlassian.net/browse/CDAP-19638)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -136,7 +136,7 @@
     <dependency>
       <groupId>io.cdap.plugin</groupId>
       <artifactId>hydrator-common</artifactId>
-      <version>${hydrator.version}</version>
+      <version>${cdap.plugin.version}</version>
```
Comment (fernst) [inline pom.xml]: Why is this change needed? In general, we use `hydrator.version` to refer to the version of the parent hydrator plugin artifact

File: src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java [hunk 2]
```diff
@@ -77,6 +80,9 @@ public class BaseSalesforceConfig extends ReferencePluginConfig {
   @Nullable
   private String loginUrl;
 
+  private final String fqnPrefix = "salesforce://";
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: final variable names should be `UPPERCASE_SEPARATED_BY_UNDERSCORES`

File: src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java [hunk 3]
```diff
@@ -125,6 +131,28 @@ public String getLoginUrl() {
...
+  }
+
+  /**
+   * Get fully-qualified name (FQN) for a Salesforce object (FQN format: salesforce://prod/orgId.mySobject).
+   *
+   * @return String fqn
+   */
+  public String getFQN(String orgId, String sObject) {
+    String firstFQNPart = loginUrl.equals(defaultLoginUrl) ? "prod" : "sandbox";
+    return fqnPrefix + firstFQNPart + "/" + orgId + "." + sObject;
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: Nitpick: use String.format to make the format clearer to understand.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 4]
```diff
@@ -103,17 +106,25 @@ public void prepareRun(BatchSourceContext context) throws ConnectionException {
...
-      recordLineage(context, config.referenceName, schema,
-                    "Read", "Read from Salesforce MultiObjects.");
+    String orgId = "unknown";
+    try {
+      orgId = config.getOrgId();
+    } catch (ConnectionException exception) {
+      collector.addFailure("Unable to get organization Id.", "Ensure Credentials are correct.");
     }
-     */
+    Asset asset = Asset.builder(config.getReferenceNameOrNoarmalizedFQN(orgId, sObjectNameField))
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java]: Typo: `Noarmalized` should be `Normalized`

**Review by fernst — CHANGES_REQUESTED** [review]
Added some comments/questions

File: src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java (hunk 2 above)
Comment (mrahanjam) [reply src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: I had it like that at first, but checkstyle failed when building the module with `Name 'FQN_PREFIX' must match pattern '^[a-z][a-zA-Z0-9]*$'`. I ended up removing the constant as it is not really needed when I used String.format.

Comment (fernst) [reply src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: Oh, you need to make this a static variable. Forgot to mention that.

Comment (fernst) [reply src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: ` private static final String FQN_PREFIX = "salesforce://";`

File: src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java [hunk 5]
```diff
@@ -77,6 +80,8 @@ public class BaseSalesforceConfig extends ReferencePluginConfig {
   @Nullable
   private String loginUrl;
 
+  private final String defaultLoginUrl = "https://login.salesforce.com/services/oauth2/token";
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: make this `private static final`

Comment (mrahanjam) [reply src/main/java/io/cdap/plugin/salesforce/plugin/BaseSalesforceConfig.java]: Done

**Review by fernst — APPROVED** [review]
